src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/MtAmazonDynamoDbBySharedTable.java

```diff
@@ -730,13 +735,15 @@ public CreateBackupResult createBackup(CreateBackupRequest createBackupRequest)
                 "Multitenant backups cannot backup individual tables, table-name arguments are disallowed");
             backupManager.get().createBackup(createBackupRequest);
 
-            ExecutorService executorService = Executors.newFixedThreadPool(mtTables.keySet().size());
+            // list all physical tables belonging to this shared table client
+            Collection<String> origPhysicalTables = new AmazonDynamoDbAdminUtils(getAmazonDynamoDb()).listTables()
+                .stream().filter(this::isPhysicalTable).collect(Collectors.toSet());
```

Comment:
This first lists all physical tables, then filters the list to one physical table underlying this `MtAmazonDynamoDbBySharedTable` instance. Do I read this correctly? 

Larger DynamoDB accounts may have hundreds if not thousands of physical tables, so I wonder if it makes sense to rework this to avoid `AmazonDynamoDbAdminUtils#listTables()` call.

Reply:
Good point. I already have a version of `AmazonDynamoDbAdminUtils#listTables()` where you pass in a filter predicate, so I'll use that version instead.